docs/usage/validators.md

```diff
@@ -336,22 +336,26 @@ In this case you should set `check_fields=False` on the validator.
 
 Validators also work with *pydantic* dataclasses.
 
-```py test="xfail - currently decorators are messing up dataclass field defaults #5271"
+# TODO: Change this example so that it _should_ use a validator; right now it would be better off with default_factory..
```

Comment:
Let me know if I should change this test in a different way, or if I should indicate the need to change it in another way than this TODO comment.

Reply:
i agree this case should use a `default_factory`.